### DISCUSSION TITLE
Many unfound character while training a model from a previously trained model

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#1863] [General Q&A]
Hi,

I am using @thorstenMueller 's German model to train a model in French (see `--restore_path`). @thorstenMueller used gruut as phonemizer so I used it too in the config.json. But I doubt the training process is working correctly since it reports nearly as many unfound characters as there are characters in the alphabets.
`
DataLoader initialization
| > Tokenizer:
	| > add_blank: True
	| > use_eos_bos: False
	| > use_phonemes: True
	| > phonemizer:
		| > phoneme language: fr-fr
		| > phoneme backend: gruut
	| > 24 not found characters:
	| > i
	| > m
	| > a
	| > f
	| > y
	| > d
	| > t
	| > ɑ
	| > ̃
	| > p
	| > ʁ
	| > k
	| > e
	| > s
	| > ɛ
	| > v
	| > w
	| > ʒ
	| > ə
	| > ɥ
	| > z
	| > ɔ
	| > b
	| > ø
| > Number of instances : 558
 | > Preprocessing samples
 | > Max text length: 169
 | > Min text length: 27
 | > Avg text length: 91.55555555555556
 | 
 | > Max audio length: 151839.0
 | > Min audio length: 24789.0
 | > Avg audio length: 89414.98566308244
 | > Num. instances discarded samples: 0
 | > Batch group size: 0.

`
I read different older posts telling that espeak should be favored instead of gruut but here @thorstenMueller 's model used gruut so I am not sure  I can / should switch to espeak.

My config.json phoneme part includes (the characters field was given by the command find_unique_chars.py) :
`
"use_phonemes": true,
    "phoneme_language": "fr-fr",
    "compute_input_seq_cache": true,
    "text_cleaner": "basic_cleaners",
    "phonemizer": "gruut",
    "enable_eos_bos_chars": false,
    "test_sentences_file": "",
    "phoneme_cache_path": "/home//Caraduf/Phonemes_cache",
    "characters":{
	"characters_class": "TTS.tts.utils.text.characters.IPAPhonemes",
        "vocab_dict": null,
        "pad": "<PAD>",
        "eos": "<EOS>",
        "bos": "<BOS>",
        "blank": "<BLNK>",
        "characters": "ABCDEFHIJLMNOPQRSTUVabcdefghijklmnopqrstuvxyzÀÇÉàâçèéêîïôùœ",
        "punctuations":" ',-.:;",
        "phonemes":"null",
        "is_unique": false,
        "is_sorted": true
    },
`
Should I have to worry about those unfound characters ? What should I do to add those missing chars ?

Thanks in advance for your help

**Comment by thorstenMueller** [comment]
Indeed the `VITS` model trained by nice @domcross is `gruut` based. I've trained a DDC model which will be released soon and this is `espeak` based. So if you have some time maybe using the espeak DDC model soon might be worth a try.

**Reply by Ca-ressemble-a-du-fake** [reply]
Thank you @thorstenMueller for the piece of information. Do you mean that using gruut based VITS models is discouraged because otherwise it leads to the issue I posted ?  Is it easy to train DDC compared to VITS ?

**Reply by thorstenMueller** [reply]
The `DDC` model will exist in addition to the `VITS` model. The error you see is because of different character set. I'm not sure if this works to finetune a model with a different character set. Maybe @WeberJulian or @erogol can give more info on that. VITS/DDC do not differ on training effort. It's more a comparison on quality and compute time to synthesize. And VITS has the vocoder already included.

**Reply by lexkoro** [reply]
You can switch to espeak, should be no problem. Additionally you will have to add the phonemes used in the French language to the config file.

**Reply by Ca-ressemble-a-du-fake** [reply]
Thank you again @thorstenMueller and @lexkoro . I tried to run the script `find_unique_phonemes` but it always complains of "phoneme language must be defined in config". Of course it is defined in config.json "phoneme_language" : "fr-fr" (also tried with "fr") and I also tried to add it to my custom formatters (caraduf) as "language" but it is not seen. Is there a way to run it ?

**Reply by Ca-ressemble-a-du-fake** [reply]
@thorstenMueller If I added to your character list my special French characters would it be work ?
